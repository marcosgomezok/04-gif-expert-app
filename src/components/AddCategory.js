import React,{useState} from 'react'
import PropTypes from 'prop-types'
export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); //useState () tiene un error de indefinido si no recibe nada, usar ''
    const handleInputChange = (e) => {
       // console.log(e.target.value)
        setInputValue(e.target.value)
        console.log('handleInputChange llamado')
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        console.log('handleSubmit',inputValue);

        if(inputValue.trim().length>2){
            setCategories(cats =>[ inputValue,...cats]);
            setInputValue('');
        }

        
    }
    
    return (
    
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
        <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
            />
        </form>    
 
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}