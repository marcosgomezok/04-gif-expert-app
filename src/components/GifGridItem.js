import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({id,title,url}) => {

    //console.log(id,title,url);
    return (
        <div className='card animate__animated animate__fadeIn'>
          <img src={url} alt={title}/>
             <p>{title}</p>
        </div>
    )
}
//  {img.title} 

/* 1. Enzyme
   2. Enzyme to json
   3. debe de mostrar el componente correcto
    shallow
    wrapper
    wrapper .toMatchSnapshot()
*/

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  
}