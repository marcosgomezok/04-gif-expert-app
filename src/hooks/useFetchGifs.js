import {useState, useEffect} from 'react'
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data:[],
        loading: true
    });

    useEffect(()=>{
        getGifs(category)
            .then(imgs =>{

                setTimeout(()=>{
                    //console.log(imgs)
                    setState({
                        data: imgs,
                        loading: false
                    });

                },1000);
  
            });
    },[category])

/*     setTimeout(()=>{
        setState({
            data:[1,2,3,4,5,6,7],
            loading:false
        })



    },3000); */
    
    
    
    
    return state; //{data:[], loading: true};


}
