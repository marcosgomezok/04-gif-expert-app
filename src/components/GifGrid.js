//import React,{useState, useEffect} from 'react'
//import { getGifs } from '../helpers/getGifs'; 
//el import de arriba se usa en el custom hook
import { GifGridItem } from './GifGridItem';
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


//    const [images, setImages] = useState([]);
//    const state = useFetchGifs();
    const {data:images,loading} = useFetchGifs(category);


    return (
        <>
        <h3 className="animate__animated animate__fadeIn">{category}</h3>
        {loading && <p>Loading</p>}
        <div className="card-grid animate__animated animate__flash">

            {
                images.map(img => (
                <GifGridItem key={img.id}
                {...img}
                />
                ))
            }
        </div> 
        </>
    )
}
GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}

    //const [count, setCount] = useState(0)

    // getGifs();
/*

    useEffect(()=>{
        getGifs(category)
            .then(setImages);
    },[category]) */

//.then(imgs => setImages(imgs))


/*     const getGifs = async() =>{

   

        const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=dQaadwLZwFULzd2dI2vMmTfZdGI3SMuj`
        
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gifs = data.map(img =>{
            return{
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url,
            }
        })
        console.log(gifs) 
        setImages(gifs);
    } */

    /* <h3>{count}</h3>
<button onClick={ () => setCount ( count + 1 ) }></button> */

//images.map(img => (
//sin desestructuracion linea 53
/* 
<ol>
{
    images.map(({id,title}) => (
                <li key={id}>{title}</li>
    ))
}
</ol>
 */

 /* <div className="card-grid">

            {
                images.map(img => (
                <GifGridItem key={img.id}
                {...img}
                />
                ))
            }
        </div> */