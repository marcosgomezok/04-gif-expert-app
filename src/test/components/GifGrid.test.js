import '@testing-library/jest-dom';//ayuda con el autocompletado
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')



describe('Pruebas en el <GifGrid />', () => {
    
    const category = "One Punch"
    test('debe de mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        
        expect(wrapper).toMatchSnapshot();

    })
    
    test('debe de mostrar items cuando se cargan imagentes useFetch', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquie/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquie/cosa.jpg',
            title: 'Cualquier cosa'
        }]

     

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        
        


    })



  
    

})
