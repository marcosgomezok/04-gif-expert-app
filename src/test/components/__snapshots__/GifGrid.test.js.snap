// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__flash"
  />
</Fragment>
`;
